prog/dftb+/lib_dftbplus/hsdhelpers.F90

```diff
@@ -72,12 +73,26 @@ subroutine doPostParseJobs(hsdTree, parserFlags)
     !> Parser specific settings in the output
     type(TParserFlags), intent(in) :: parserFlags
 
-    type(fnode), pointer :: root
+    type(fnode), pointer :: root, node
+    type(fnodeList), pointer :: ignoreList, nodeList
 
     call getChild(hsdTree, rootTag, root)
 
+    ! Sets a list of nodes that will be ignored even if not processed
+    ! Logic: warnUnprocessed will check each node and its parent with ignoreList 
+    !        Hence setting "MaxAngularMomentum" will allow ignoring all 
+    !        its children elements  
+    nullify(ignoreList)
+    node => createElement("")
+    call setNodeName(node,"MaxAngularMomentum")
```

Comment:
I think, the nodes, which should not trigger an unprocessed warning, should be marked during the parsing in `parser.F90` by setting a specific hsd-attribute (e.g. ``hsdIgnoreUnproc``), rather than here. Setting them here would mix the generic parsing functions and specific DFTB+ related parsing issues. This routine is used by several hsd-parser (e.g. modes, waveplot, etc.) where a ``MaxAngularMomentum`` node does not appear. Also nodes added here won't have the proper path information (what, if there are nodes with the same name on different levels, and only one of them should be ignored...).